plenum/server/node.py

```diff
@@ -2130,6 +2125,161 @@ def _is_there_pool_ledger(self):
         # TODO isinstance is not OK
         return isinstance(self.poolManager, TxnPoolManager)
 
+    def process_message_req(self, msg: MessageReq, frm):
+        # Assumes a shared memory architecture. In case of multiprocessing,
+        # RPC architecture, use deques to communicate the message and node will
+        # maintain a unique internal message id to correlate responses.
+        resp = None
+        msg_type = msg.msg_type
+        if msg_type == LEDGER_STATUS:
```

Comment:
The bottom line is that as the requesting criteria becomes complex, it can sometimes be impossible to statically validate `params`. `MessageReq` is fundamentally different that other messages. It is a query for other messages. This method is shorter now